relation_engine_server/utils/ensure_specs.py

```diff
@@ -0,0 +1,260 @@
+        view_local = load_json_yaml(view_spec_path)
+        match = False
+        for view_server in all_views_server:
+            if view_local.items() <= view_server.items():
+                match = True
+                break
+        if match is False:
+            failed_specs.append(view_local)
+
+    if failed_specs:
+        print_failed_vs_server("views", failed_specs, all_views_server)
+    else:
+        print("All view specs ensured")
+
+    return get_names(failed_specs, "views"), failed_specs
+
+
+def ensure_analyzers():
+    """
+    Returns tuple
+    First item is list of failed view names, e.g.,
+    [
+       "icu_tokenize/text"
+    ]
+    Second item is list of failed specs, e.g.,
+    [
+        {"name": "icu_tokenize", "type": "text", ...}
+    ]
+    """
+    all_analyzers_server = arango_client.get_all_analyzers()
+    mod_obj_literal(all_analyzers_server, str, excise_namespace)
+    analyzer_spec_paths = get_schema_type_paths("analyzer")
+
+    failed_specs = []
+    for analyzer_spec_path in analyzer_spec_paths:
+        print(f"Ensuring analyzer {analyzer_spec_path}")
+        analyzer_local = load_json_yaml(analyzer_spec_path)
+        for analyzer_server in all_analyzers_server:
+            match = False
+            if analyzer_local.items() <= analyzer_server.items():
+                match = True
+                break
+        if match is False:
+            failed_specs.append(analyzer_local)
+
+    if failed_specs:
+        print_failed_vs_server("analyzers", failed_specs, all_analyzers_server)
+    else:
+        print("All analyzer specs ensured")
+
+    return get_names(failed_specs, "analyzers"), failed_specs
+
+
+def ensure_all():
+    """
+    Return names of failed specs if any, e.g.,
+    {
+        "indexes": [
+        ],
+        "views": [
+            "Coumpounds/arangosearch",
+            "Reactions/arangosearch",
+        ],
+        "analyzers": [
+            "icu_tokenize/text",
+        ],
+    }
+    """
+    failed_indexes_names, _ = ensure_indexes()
+    failed_views_names, _ = ensure_views()
+    failed_analyzers_names, _ = ensure_analyzers()
+
+    return {
+        "indexes": failed_indexes_names,
+        "views": failed_views_names,
+        "analyzers": failed_analyzers_names,
+    }
+
+
+def get_names(specs, schema_type):
+    """
+    Given views/analyzers/collections, collate names using required properties
+    """
+    names = []
+    if schema_type in ["views", "analyzers"]:
+        for spec in specs:
+            names.append(f"{spec['name']}/{spec['type']}")
+    elif schema_type in ["indexes"]:
+        for coll_name, indexes in specs.items():
+            for index in indexes:
+                names.append(f"{coll_name}/{index['type']}/{index['fields']}")
+    else:
+        raise RuntimeError(f'Unknown schema type "{schema_type}"')
+    return names
+
+
+def print_failed_vs_server(schema_type, failed_specs, server_specs):
```

Comment:
At least this will print out useful debugging info, but it may not show in `curl`?

Reply:
should this fail message be returned to the user or just print debugging info on the server?

Reply:
This is really rough ATM. It prints out all the failed specs and all the server specs. It needs work.

Reply:
Revised.